### PR TITLE
Update Jetpack cloud to be more in line with calypso

### PR DESCRIPTION
**Pull request by enejb**
This PR removes cloud specific layout controllers and replaced them with calypso specific ones.

In https://github.com/Automattic/wp-calypso/pull/40022 we started using the calypso specific entry point. 

This PR continuous this work by removing the jetpack cloud specific functions that are replicas of what is already available in calypso. 

Before:
Screen Shot 2020-03-12 at 2 21 57 PM

After:
Screen Shot 2020-03-12 at 2 39 48 PM

#### Changes proposed in this Pull Request
* Removes the Jetpack Cloud specific entrypoint functions.

#### Testing instructions
* Load up calypso `npm start`. Does it still look the same as before? Any regressions? 
* Load up jetpack cloud with `npm run start-jetpack-cloud` 
- Notice that the masterbar is always what you expect during the loading process. 
- Notice the sidebar has the sites selector. 
- Notice that when you visit a section without the site selected you end up seeing the site selector.

### CODE REVIEW

File: client/layout/index.jsx [hunk 1]
```diff
@@ -100,6 +101,19 @@ class Layout extends Component {
 		// intentionally don't remove these in unmount
 	}
 
+	renderMasterbar() {
+		const MasterbarComponent = config.isEnabled( 'jetpack-cloud' )
```
Comment (rcoll) [inline client/layout/index.jsx]: Not a blocker at all, but the way this is written seems weird. It's crafty, yes, and I commend it. However, it's hard to understand what's going on at a glance. Prefer something like 

`return config.isEnabled( 'jetpack-cloud' )
	? <JetpackCloudMasterbar ...>
	: <MasterbarLoggedIn ...>`

File: client/my-sites/controller.js [hunk 2]
```diff
@@ -87,9 +89,12 @@ function createNavigation( context ) {
 		basePath = sectionify( context.pathname );
 	}
 
+	const SidebarComponent = config.isEnabled( 'jetpack-cloud' ) ? JetpackCloudSidebar : Sidebar;
+
 	return (
 		<NavigationComponent
 			path={ context.path }
+			sidebar={ <SidebarComponent path={ context.path } siteBasePath={ basePath } /> }
```
Comment (rcoll) [inline client/my-sites/controller.js]: Again, this is a bit hard to follow. Not a blocker, but a more classic approach might be better.

**Review by rcoll — APPROVED** [review]
I only have a few comments inline about readability. Otherwise this looks fine. No blockers. 

I tested calypso hard, but it would be so easy to miss a regression. Others should test also.

File: client/landing/jetpack-cloud/components/sidebar/index.jsx [hunk 3]
```diff
@@ -48,9 +48,11 @@ class JetpackCloudSidebar extends Component {
 		const { selectedSiteSlug, translate } = this.props;
 
 		return (
-			<Sidebar>
-				<CurrentSite />
+			<Sidebar
+				className="is-jetpack-cloud-sidebar" // eslint-disable-line wpcalypso/jsx-classname-namespace
```
Comment (rcanepa) [inline client/landing/jetpack-cloud/components/sidebar/index.jsx]: What if we use use `className="sidebar sidebar--jetpack-cloud"`? The `eslint` comment can be deleted and `"sidebar--jetpack-cloud"` better conveys this is a modifier of the original sidebar.

File: client/landing/jetpack-cloud/components/sidebar/style.scss [hunk 4]
```diff
@@ -1,42 +1,45 @@
...
-	padding-top: 14px;
-	padding-bottom: 14px;
-	font-weight: 600;
-
-	.selected &,
-	.selected &:hover {
-		background-color: rgba( var( --color-sidebar-menu-selected-background-rgb ), 0.12 );
-	}
-}
+.is-jetpack-cloud-sidebar {
```
Comment (rcanepa) [inline client/landing/jetpack-cloud/components/sidebar/style.scss]: According to what I proposed above, this line would have to be `.sidebar--jetpack-cloud`.

**Review by rcanepa — APPROVED** [review]
Hey @enejb !
I followed your testing instructions and everything looks fine.

> Load up calypso. Does it still look the same as before? Any regressions?

I didn't find any regression and it looks the same as before. I compared it against staging as well. No errors in the console either.

> Notice that the masterbar is always what you expect during the loading process.

This is fine.

> Notice the sidebar has the sites selector.

This works fine as well.
    
> Notice that when you visit a section without the site selected you end up seeing the site selector.

This works as well.

File: client/my-sites/controller.js (hunk 2 above)
Comment (enejb) [reply client/my-sites/controller.js]: Can you explain the classic approach?

File: client/layout/index.jsx (hunk 1 above)
Comment (enejb) [reply client/layout/index.jsx]: The reason why it is written this way is to emphasis the differences and similarities. 
The only difference currently is the is the component function, the props (function arguments) are the same. 

Written the other way the emphasis is on both. (component and props) 

I am happy to rewrite it, but I think this approach conveys more meaning. I am happy to change it, but in my previous review the feedback I got from @tyxla was to write it this way.

File: client/my-sites/controller.js (hunk 2 above)
Comment (enejb) [reply client/my-sites/controller.js]: Here is a good explanation of the pattern used https://www.robinwieruch.de/react-pass-props-to-component#how-to-pass-components-as-props

File: client/landing/jetpack-cloud/components/sidebar/index.jsx [hunk 5]
```diff
@@ -48,9 +48,7 @@ class JetpackCloudSidebar extends Component {
 		const { selectedSiteSlug, translate } = this.props;
 
 		return (
-			<Sidebar
-				className="is-jetpack-cloud-sidebar" // eslint-disable-line wpcalypso/jsx-classname-namespace
-			>
+			<Sidebar className="sidebar sidebar__jetpack-cloud">
```
Comment (tyxla) [inline client/landing/jetpack-cloud/components/sidebar/index.jsx]: The `sidebar` class will be [added anyway](https://github.com/Automattic/wp-calypso/blob/d77f51b8a4cd5468e062028a8e0a8a70e3f2fe04/client/layout/sidebar/index.jsx#L15) so it could be removed from here.

File: client/landing/jetpack-cloud/components/sidebar/index.jsx [hunk 6]
```diff
@@ -11,7 +11,7 @@ import { localize } from 'i18n-calypso';
  */
 import { getSelectedSiteSlug } from 'state/ui/selectors';
 import config from 'config';
-import CurrentSite from '../current-site';
+import CurrentSite from 'my-sites/current-site';
```
Comment (tyxla) [inline client/landing/jetpack-cloud/components/sidebar/index.jsx]: 👍

File: client/landing/jetpack-cloud/components/sidebar/style.scss [hunk 7]
```diff
@@ -1,42 +1,45 @@
...
-	font-weight: 600;
+.sidebar__jetpack-cloud {
 
-	.selected &,
-	.selected &:hover {
-		background-color: rgba( var( --color-sidebar-menu-selected-background-rgb ), 0.12 );
-	}
-}
+	.sidebar__menu-link {
+		// padding-top: 14px;
```
Comment (tyxla) [inline client/landing/jetpack-cloud/components/sidebar/style.scss]: Perhaps we should just remove those instead of commenting them out? Or perhaps we could add a clarifying comment, explaining why they're currently commented out?

File: client/layout/index.jsx (hunk 1 above)
Comment (tyxla) [reply client/layout/index.jsx]: Yup, I agree with the approach here. The only downside is that we're always importing both components, even though we're going to use just one of them, but it's a compromise we should be fine with, given that the masterbar is immediately visible and is the first visible thing at the top of the page.

File: client/my-sites/controller.js [hunk 8]
```diff
@@ -87,9 +89,11 @@ function createNavigation( context ) {
 		basePath = sectionify( context.pathname );
 	}
 
+	const SidebarComponent = config.isEnabled( 'jetpack-cloud' ) ? JetpackCloudSidebar : Sidebar;
+
 	return (
 		<NavigationComponent
-			path={ context.path }
+			sidebar={ <SidebarComponent path={ context.path } siteBasePath={ basePath } /> }
```
Comment (tyxla) [inline client/my-sites/controller.js]: I wonder if we could `AsyncLoad` the sidebar component here. Just so we wouldn't import both components when only one of them will be used at any given time. This could have a positive impact on the bundle size, too.

File: client/my-sites/navigation/index.jsx [hunk 9]
```diff
@@ -26,7 +25,7 @@ class MySitesNavigation extends React.Component {
 					siteBasePath={ this.props.siteBasePath }
 					onClose={ this.preventPickerDefault }
 				/>
-				<Sidebar path={ this.props.path } siteBasePath={ this.props.siteBasePath } />
+				{ this.props.sidebar }
```
Comment (tyxla) [inline client/my-sites/navigation/index.jsx]: I wonder what's the benefit of specifying the logic of which component to load here in the parent component, vs doing it here?

File: client/landing/jetpack-cloud/components/sidebar/style.scss (hunk 7 above)
Comment (enejb) [reply client/landing/jetpack-cloud/components/sidebar/style.scss]: yeah I think I was testing something. I will add it back for now. So that it is more consistant with the previous jetpack cloud design.

File: client/landing/jetpack-cloud/components/sidebar/index.jsx (hunk 5 above)
Comment (enejb) [reply client/landing/jetpack-cloud/components/sidebar/index.jsx]: :+1:

File: client/my-sites/controller.js (hunk 8 above)
Comment (enejb) [reply client/my-sites/controller.js]: Will try this in a different PR.

File: client/my-sites/navigation/index.jsx (hunk 9 above)
Comment (enejb) [reply client/my-sites/navigation/index.jsx]: I think the benefit is that we are keeping the component simpler. Not sure what other benefit there is.

Comment (tyxla) [reply client/my-sites/navigation/index.jsx]: I think an extra layer of indirection is only complicating things. But happy to keep it as-is if you feel this is better.

File: client/landing/jetpack-cloud/components/sidebar/style.scss (hunk 4 above)
Comment (enejb) [reply client/landing/jetpack-cloud/components/sidebar/style.scss]: 👍

File: client/landing/jetpack-cloud/components/sidebar/index.jsx (hunk 3 above)
Comment (enejb) [reply client/landing/jetpack-cloud/components/sidebar/index.jsx]: :+1:

**Review by tyxla — APPROVED** [review]
LGTM 🚢 

Would be great if we could `AsyncLoad` the sidebar (and maybe the masterbar, if it doesn't hurt the UI of course). Including both is what seems to be causing the increase in the main entry point bundle size:

```
name        parsed_size           gzip_size
entry-main      +7738 B  (+0.5%)    +2693 B  (+0.7%)
```
